### ISSUE TITLE
Leading space is mangled when indenting with Tab

### ISSUE THREAD

**Issue by martinvonwittich** [eight04/textarea-plus#3]
To reproduce the issue, put the following text into a `textarea`, select the lines and press Tab:

```
not indented
 indented with one space
  indented with two spaces
   indented with three spaces
    indented with four spaces
     indented with five spaces
```

What I would have expected was this:

```
	not indented
	 indented with one space
	  indented with two spaces
	   indented with three spaces
	    indented with four spaces
	     indented with five spaces
```

What I get is this:

```
	not indented
	indented with one space
	indented with two spaces
	indented with three spaces
		indented with four spaces
		indented with five spaces
```

I understand that this may be intentional, but I found it to be counterproductive when working with Markdown (that's what I use Textarea Plus mainly for). I often copy-paste command output from a console into my Markdown documents and then indent them with Tab, but by doing that, I lose the existing leading space contained in the command output.

Could this behaviour be changed or made configurable?

**Comment by eight04** [comment]
Agreed. Currently, the indentation would be normalized to tab before indent/unindent, but additional spaces are stripped:

https://github.com/eight04/textarea-plus/blob/2457392fcc2c85545bfa61cdd0eecad9881aa55d/textarea-plus.user.js#L151
https://github.com/eight04/textarea-plus/blob/2457392fcc2c85545bfa61cdd0eecad9881aa55d/textarea-plus.user.js#L191

It should keep additional spaces.

**Comment by eight04** [comment]
Fixed in 3.0.0.
